src/Microsoft.Identity.Client/Internal/Requests/BaseRequest.cs

```diff
@@ -45,7 +45,7 @@ internal abstract class BaseRequest
         protected TokenResponse Response;
         protected AccessTokenCacheItem AccessTokenItem;
 
-        internal RequestContext RequestContext { get; set; }
+        internal RequestContext RequestContext { get; set; } = new RequestContext(Guid.Empty);
```

Comment:
this should never be set to Guid.Empty. It will always be created in public and confidential client class